CombineDemo/CombineDemo/Modules/PokemonList/PokemonTableViewCell.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  PokemonTableViewCell.swift
+//  CombineDemo
+//
+//  Created by Manish Rathi on 19/08/2023.
+//
+
+import UIKit
+import Combine
+
+class PokemonTableViewCell: UITableViewCell {
+    lazy var imageLoader: ImageLoading = ImageLoader()
+    lazy var cancellables: Set<AnyCancellable> = []
+
+    private enum Constants {
+        static let height: CGFloat = 100
+        static let spacing: CGFloat = 8
+    }
+
+    lazy var pokemonImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.layer.cornerRadius = Constants.height / 2
+        imageView.layer.borderWidth = 0.25
+        imageView.layer.borderColor = UIColor.link.cgColor
+        imageView.contentMode = .scaleAspectFill
+        imageView.clipsToBounds = true
+        return imageView
+    }()
+
+    lazy var pokemonNameLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = .label
+        return label
+    }()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        setupViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func prepareForReuse() {
+        super.prepareForReuse()
+
+        imageView?.image = nil
+        imageLoader.cancelImageLoading()
```

Comment:
I don't think that the imageLoader is the owner of requests. You decided to have cancellable in the cell. You can set it to empty and requests will be cancelled

Reply:
Good point! that is indeed true too!! Making cancellable empty will cancel the request too. but i was thinking, `imageLoader.cancelImageLoading()` will be more explicit and readable for canceling the request. 

To be honest, i don't know the best practice people follow in Combine!! 😇

Reply:
In reactive programming is everything is cancellable through that variable. You no need to think how to cancel request or processing or anything. Just clean the disposable bag/cancellable. That's the only pattern to not think how it works inside or what is the public API, but you always can cancel it.

Reply:
Also the more important that some functionality could use another one and another one. So we don't know the chain of the process, but we can cancel everything if we need